server/src/main/java/org/elasticsearch/action/bulk/TransportSimulateBulkAction.java

```diff
@@ -351,7 +351,7 @@ private Collection<String> validateUpdatedMappings(
             .putMapping(new MappingMetadata(updatedMappings))
             .build();
         Engine.Index result = indicesService.withTempIndexService(originalIndexMetadata, indexService -> {
-            indexService.mapperService().merge(updatedIndexMetadata, MapperService.MergeReason.MAPPING_UPDATE);
+            indexService.mapperService().merge(updatedIndexMetadata, MapperService.MergeReason.INDEX_TEMPLATE);
```

Comment:
Hmm… I'm worried we kind of want both behaviors right? The `MAPPING_UPDATE` one is if we were to update the mapping via the API, while the `INDEX_TEMPLATE` one is for updating the template. Do we know for sure what the intent is on the user's side?

Reply:
My thinking was that if a user wanted to simulate this change, they'd be doing it through an index-template-like update rather than a mapping update, since this change just fails with a mapping update. This API is meant to simulate what would happen if the index had this mapping change, rather than what would happen if I attempted to make this mapping change.